tools/check_lint_rules_identity/lib/check_lint_rules_identity.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:check_lint_rules_identity/src/diff_path_list_file.dart';
+import 'package:check_lint_rules_identity/src/services/diff_version_service.dart';
+import 'package:check_lint_rules_identity/src/models/exit_status.dart';
+import 'package:check_lint_rules_identity/src/services/identity_verification_service.dart';
+import 'package:check_lint_rules_identity/src/lint_rules_dir.dart';
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:meta/meta.dart';
+import 'package:riverpod/riverpod.dart';
+
+Future<ExitStatus> run(List<String> argument) async {
+  final fileSystem = LocalFileSystem();
+
+  final diffPathListFilePath = argument.firstOrNull;
+  if (diffPathListFilePath == null ||
+      !fileSystem.isFileSync(diffPathListFilePath)) {
+    throw ArgumentError(
+        'The first argument must be a valid path to an existing file.');
+  }
+
+  final lintRulesDirPath = argument.elementAtOrNull(1);
+  if (lintRulesDirPath == null ||
+      !fileSystem.isDirectorySync(lintRulesDirPath)) {
+    throw ArgumentError(
+        'The input value must be a valid path to an existing directory.');
+  }
+
+  final File diffPathListFile = fileSystem.file(diffPathListFilePath);
+  final Directory lintRulesDir = fileSystem.directory(lintRulesDirPath);
+
+  final container = ProviderContainer(
+    overrides: [
+      lintRulesDirProvider.overrideWithValue(lintRulesDir),
+      diffPathListFileProvider.overrideWithValue(diffPathListFile),
+    ],
+  );
+  try {
+    final exitStatus = await checkLintRulesIdentity(container);
+    return exitStatus;
+  } finally {
+    container.dispose();
+  }
+}
+
+@visibleForTesting
+Future<ExitStatus> checkLintRulesIdentity(ProviderContainer container) async {
+  final diffVersionService = container.read(diffVersionServiceProvider);
+  final dartIdentityVerificationService =
+      container.read(dartIdentityVerificationServiceProvider);
+  final flutterIdentityVerificationService =
+      container.read(flutterIdentityVerificationServiceProvider);
+
+  try {
+    final diffVersions = diffVersionService.getDiffVersion();
+    final isDartIdentity = await diffVersions.dart
+        .asyncEvery(dartIdentityVerificationService.isIdenticalLintRule);
+    final isFlutterIdentity = await diffVersions.flutter
+        .asyncEvery(flutterIdentityVerificationService.isIdenticalLintRule);
+
+    if (isDartIdentity && isFlutterIdentity) {
+      return ExitStatus.success;
+    }
+    return ExitStatus.warnings;
+  } on Exception catch (e) {
+    print(e);
+    return ExitStatus.errors;
+  }
+}
+
+extension AsyncEvery<E> on Set<E> {
+  Future<bool> asyncEvery(Future<bool> Function(E element) test) async {
+    for (E element in this) {
+      if (!await test(element)) return false;
+    }
+    return true;
+  }
```

Comment:
![memo-badge](https://img.shields.io/badge/review-memo-lightgrey)
If there is even one difference, the subsequent processes will not be executed.

Reply:
![ask-badge](https://img.shields.io/badge/review-ask-yellowgreen.svg)
If you want to output logs in more detail, I think it might be better to check all the differences. What do you think?

Reply:
First, change asyncEvery to check all elements.
fb350ea8bb2e9049d7a8cb6eb77ad82acd3bf72a

But after test function returned false, it was omitted for some reason.
db58cee13de95915ebba6947b4c19ee123f12748

Reply:
> Rename diffPathListFileProvider to versionPathsFileProvider. https://github.com/yumemi-inc/flutter-yumemi-lints/commit/663b5624b6a6abcc540915504dc4b7ebcd76dc78

Because short-circuit evaluation. ([wiki](https://ja.wikipedia.org/wiki/%E7%9F%AD%E7%B5%A1%E8%A9%95%E4%BE%A1))